firmware/pico/src/app/hal.hh

```diff
@@ -0,0 +1,44 @@
+#pragma once
+
+#include <cstdint>
+
+uint64_t get_time_since_boot_us();
+uint32_t get_time_since_boot_ms();
+
+/**
+ * Singleton class to manage the systick timer in the ARM cores
+ * @attention Since this is a singleton and uses per core hardware, this should only be constructed from Core 0 to maintain consistency
+ * @see https://documentation-service.arm.com/static/5f8ff05ef86e16515cdbf826 section B3.3
+ * 
+ */
+class ClockSource {
+   public:
+    static ClockSource& instance();
+    ~ClockSource();
+
+    /**
+     * Creates a composite timestamp using the current value of the SysTick timer (running at 125MHz) and the SysTick
+     * wrap counter to simulate a timer running at 48MHz (which matches the frequency of the preamble detector PIO).
+     * @retval 48MHz counter value.
+     */
+    uint64_t Get48MHzTickCount();
```

Comment:
Let's add the num_bits parameter back, unless there is a good reason to remove it.